File: src/types/schema.ts
export type SchemaObjectType = 'database' | 'schema' | 'table' | 'view' | 'column';

export interface ColumnMetadata {
  name: string;
  dataType: string;
  maxLength?: number;
  precision?: number;
  scale?: number;
  isNullable: boolean;
  isPrimaryKey: boolean;
  isIdentity: boolean;
}

export interface TableMetadata {
  schema: string;
  name: string;
  type: 'BASE TABLE' | 'VIEW';
  columns?: ColumnMetadata[];
}

export interface DatabaseMetadata {
  name: string;
  schemas: string[];
  tables: TableMetadata[];
}

export interface SchemaTreeNode {
  id: string;
  label: string;
  type: SchemaObjectType;
  children?: SchemaTreeNode[];
  isLoaded?: boolean;
  metadata?: Record<string, unknown>;
}

File: src-tauri/src/models/schema.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DatabaseItem {
    pub name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SchemaItem {
    pub name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TableItem {
    pub schema: String,
    pub name: String,
    pub kind: String, // 'BASE TABLE' or 'VIEW'
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ColumnItem {
    pub name: String,
    pub data_type: String,
    pub max_length: Option<i32>,
    pub precision: Option<u8>,
    pub scale: Option<u8>,
    pub is_nullable: bool,
    pub is_primary_key: bool,
    pub is_identity: bool,
}
